### PR TITLE
Throttling Capability

### PR DESCRIPTION
**Pull request by battinasubodh**
Changes in the core module to handle request throttling. Thanks !

### CODE REVIEW

File: core/src/main/java/com/capitalone/dashboard/collector/ThrottleRequests.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.capitalone.dashboard.collector;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public interface ThrottleRequests {
```
Comment (satishc1) [inline core/src/main/java/com/capitalone/dashboard/collector/ThrottleRequests.java]: @battinasubodh please move to util.

File: core/src/main/java/com/capitalone/dashboard/collector/ThrottleSettings.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.capitalone.dashboard.collector;
+
+/**
+ * Bean to hold throttle settings used to throttle the number
+ * of requests made within a specific time window.
+ */
+public class ThrottleSettings {
```
Comment (satishc1) [inline core/src/main/java/com/capitalone/dashboard/collector/ThrottleSettings.java]: @battinasubodh please move to model

**Comment by subodhbattina** [comment]
@satishc1 Have made the changes. Thanks !

**Comment by satishc1** [comment]
@tabladrum LGTM

File: core/src/main/java/com/capitalone/dashboard/model/ThrottleSettings.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public int getRequestRateLimit() { return requestRateLimit; }
+
+    public void setRequestRateLimit(int requestRateLimit) { this.requestRateLimit = requestRateLimit; }
+
+    public long getWaitTime() { return waitTime; }
+
+    public void setWaitTime(long waitTime) { this.waitTime = waitTime; }
+}
```
Comment (tabladrum) [inline core/src/main/java/com/capitalone/dashboard/model/ThrottleSettings.java]: Why do we need another interface and not just a method (overrideable) method in base CollectorTask?

Comment (tabladrum) [reply core/src/main/java/com/capitalone/dashboard/model/ThrottleSettings.java]: How's this setting being injected?

**Comment by tabladrum** [comment]
@satishc1 should we be reverting this until review is complete?

**Comment by satishc1** [comment]
@tabladrum @battinasubodh a followup pr would be good.

**Pull request merged**